src/main/java/seedu/address/model/loan/Nric.java

```diff
@@ -0,0 +1,26 @@
+package seedu.address.model.loan;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Represents a Loan's NRIC number in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValid}
+ */
+public class Nric extends LoanField<String> {
+
+    public static final String MESSAGE_CONSTRAINTS =
+        "NRIC numbers should start with a S, T, F or G, followed by 7 decimal digits, then end with a letter.";
+
+    public static final Predicate<String> isValid = test ->
+        test.matches("[STFGstfg]\\d{7}[A-Za-z]");

```

Comment:
Last letter is in `[A-JZa-jz]`.

Reply:
Propose change to `[STFGstfg]\d{7}[A-JZa-jz]`

Reply:
Also, Singapore NRIC's last digit is a checksum. We should implement that check sum as well.